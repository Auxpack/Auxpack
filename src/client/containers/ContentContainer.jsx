import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Overview from '../content/containers/Overview.jsx';
import BuildData from '../content/containers/BuildData.jsx';
import TreeShaking from '../content/containers/TreeShaking.jsx';
import HistoryCharts from '../content/containers/HistoryCharts.jsx';

const ContentContainer = (props) => {
    const { build, activeBuild, handleInc, handleDec } = props;
    return (
        <React.Fragment>
            <Switch>
                <Route
                    exact path="/"
                    render={() => (
                        <Overview
                         build={build} 
                         activeBuild={activeBuild} 
                        />
                    )}
                />
                <Route
                    exact path="/build"
                    render={() => (
                        <BuildData
                            build={build}
                            activeBuild={activeBuild}
                            handleDec={handleDec}
                            handleInc={handleInc}
                        />
                    )}
                />
                <Route
                    exact path="/treeshaking"
                    render={() => (
                        <TreeShaking
                        build={build}
                        activeBuild={activeBuild}
                        />
                    )}
                />
                <Route
                    exact path="/history"
                    render={() => (
                        <HistoryCharts
                        build={build}
                        />
                    )}
                />
            </Switch>
        </React.Fragment>
    );
}

export default ContentContainer;