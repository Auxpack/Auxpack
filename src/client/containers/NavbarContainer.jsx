import React from 'react';
import BottomNavigation from '../components/BottomNavigation.jsx'
import BuildSelect from '../components/BuildSelect.jsx'

const NavbarContainer = ({build, activeBuild, selectBuild}) => {
    //navbar split up into components for the bottom navigation that changes routes and a selector for active builds
    return ( 
        <div id="nav">
            <BottomNavigation />
            <BuildSelect build={build} activeBuild={activeBuild} selectBuild={selectBuild} />
        </div>
        
     );
}
 
export default NavbarContainer;