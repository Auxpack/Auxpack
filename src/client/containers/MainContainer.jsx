import React, { useEffect, useState } from 'react'
import NavbarContainer from './NavbarContainer';
import ContentContainer from './ContentContainer'

const MainContainer = (props) => {
    const [build, setBuild] = useState([]);
    const [activeBuild, setActiveBuild] = useState(0);

    useEffect(() => {
        console.log(`fetching in USEEFFECT`)
        fetch('/getStats')
            .then(res => res.json())
            .then(data => {
                setBuild(data)
                setActiveBuild(data.length - 1)
            })
            .catch(err => console.log(err))
    }, [])



    const clickHandler = e => {
        const length = build.length;
        const i = length - e.target.getAttribute('data-build');
        setActiveBuild(length - i);
    }

    const selectBuild = e => {
        const i = parseInt(e.target.value);
        setActiveBuild(i)
    }

    const handleInc = () => {
        if (activeBuild < build.length - 1) {
            activeBuild += 1;
            setActiveBuild(activeBuild)
        }
    }

    const handleDec = () => {
        if (activeBuild > 0) {
            activeBuild -= 1;
            setActiveBuild(activeBuild)
        }
    }

    // Navbar props: build, activeBuild, selectBuild
    // ContentContainer props: build, activeBuild, all methods except selectBuild

    return (
        <React.Fragment>
            <NavbarContainer
                build={(build.length !== 0) ? build : []}
                activeBuild={activeBuild}
                selectBuild={selectBuild}
            />
            <ContentContainer
                build={(build.length !== 0) ? build : []}
                activeBuild={activeBuild}
                clickHandler={clickHandler}
                handleInc={handleInc}
                handleDec={handleDec}
            />
        </React.Fragment>
    );

}

export default MainContainer;