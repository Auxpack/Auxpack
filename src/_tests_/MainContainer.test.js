import MainContainer from '../client/containers/MainContainer.jsx';
import ContentContainer from '../client/containers/ContentContainer.jsx';
import NavbarContainer from '../client/containers/NavbarContainer.jsx';

describe('MainContainer Unit Tests', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<MainContainer />);
  });

  it('should render', () => {
    expect(wrapper);
  });

  test('MainContainer snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('MainContainer should render ContentContainer', () => {
    expect(wrapper.find(ContentContainer)).toHaveLength(1);
  });

  it('MainContainer should render NavbarContainer', () => {
    expect(wrapper.find(NavbarContainer)).toHaveLength(1);
  });
});
