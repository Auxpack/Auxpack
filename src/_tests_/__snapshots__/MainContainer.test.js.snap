// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer Unit Tests MainContainer snapshot 1`] = `
<Fragment>
  <ContentContainer
    activeBuild={0}
    build={Array []}
    clickHandler={[Function]}
    handleDec={[Function]}
    handleInc={[Function]}
  />
  <NavbarContainer
    activeBuild={0}
    build={Array []}
    selectBuild={[Function]}
  />
</Fragment>
`;
